src/Reopt/TypeInference/ConstraintGen.hs

```diff
@@ -701,13 +710,24 @@ genModule m mem = fst $ runCGenM mem $ do
   tyVars <- CGenM $ use assignTyVars
   tvsO <- CGenM $ use tyVarsOrigin
   warns <- CGenM $ use warnings
-  cstrs <- CGenM $ use constraints
+
+  -- Set to False if we _do not_ want to fresh type variables to describe
+  -- what the inferred pointers point to.
```

Comment:
Can we get a short description here of the implications of generating a fresh variable vs not?

Reply:
How about if instead of what's there now it says: If `True` then inferred pointers will be assigned type `ptr(α)` (where `α` is a fresh type variable); if `False`, inferred pointers will be assigned type `ptr(⊤)`. Importantly, the former allows us to relate learned things about `α` back to other types when possible (e.g., learning things about records/structs, possibly), whereas the latter permits no such transitive reasoning in the inference engine.

Reply:
I like it